crates/net/network/src/transactions/mod.rs

```diff
@@ -1053,22 +1060,32 @@ where
 
         // Advance all imports
         while let Poll::Ready(Some(import_res)) = this.pool_imports.poll_next_unpin(cx) {
-            match import_res {
-                Ok(hash) => {
-                    this.on_good_import(hash);
-                }
+            let import_res = match import_res {
+                Ok(res) => res,
                 Err(err) => {
-                    // if we're _currently_ syncing and the transaction is bad we ignore it,
-                    // otherwise we penalize the peer that sent the bad
-                    // transaction with the assumption that the peer should have
-                    // known that this transaction is bad. (e.g. consensus
-                    // rules)
-                    if err.is_bad_transaction() && !this.network.is_syncing() {
-                        debug!(target: "net::tx", ?err, "bad pool transaction import");
-                        this.on_bad_import(err.hash);
-                        continue
+                    debug!(target: "net::tx", ?err, "bad pool transaction batch import");
+                    continue
+                }
+            };
```

Comment:
I just realized that we might even get rid of the additional result wrapper here but can simplify this in a followup:

https://github.com/paradigmxyz/reth/blob/add3c6a3a2ae0cecd8627533bbc5688a456435ad/crates/transaction-pool/src/lib.rs#L240-L240